### PR TITLE
feat: added scroll indicator

### PR DESCRIPTION
**Pull request by Sourabh782**
## Related Issue

fixes: #417 

## Description
Added a Scroll Indicator at the top of site, with matching theme for both dark and light mode. working efficiently as we scroll through the site, also working with the "to the top button" responsively.

## Screenshots

https://github.com/user-attachments/assets/cece7077-f187-4490-9d01-02c8ec488404

Changes in code base: 
Added a ProgressBar Component

## Summary by CodeRabbit

- **New Features**
	- Introduced a `ProgressBar` component to visually indicate scroll progress on the homepage and the contribution guide page.
- **Style**
	- Adjusted the `Navbar` positioning for improved visual layout by adding a gap from the top of the viewport.

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -13,6 +14,7 @@ export default async function Home() {
 
   return (
     <>
+      <ProgressBar />
```
Comment (Sanchitbajaj02) [inline src/app/page.tsx:17]: can't we use the framer motion here? The same is implemented in:
`src/components/pages/contribute/index.tsx`

File: src/components/ProgressBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+"use client";
```
Comment (Sanchitbajaj02) [inline src/components/ProgressBar/index.tsx:1]: feels a lot of extra work since this feature is already developed

**Review by Sanchitbajaj02 — CHANGES_REQUESTED** [review]
Try using the framer motion library and the mentioned component code to implement the logic.

**Comment by Sourabh782** [comment]
Sure working on it 👍

**Comment by Sanchitbajaj02** [comment]
> @Sanchitbajaj02 hello sir, actually i am having an issue here, as u said to use framer motion library so i can use that, but the 
> src/app/page.tsx ( Where i m implementing scrollbar ) is showing an error
> 
> Screenshot 2024-08-08 184442
> 
> to use frammer motion directly i have to use "use client" which generate this error.
> 
> but if i implement it using another component ( moving scroll Indicator's code to another component ), it is working fine.
> 
> so can i implement it by making a seprate component that uses frammer motion?
> 

What you can do in this case is, modify your progressbar component using framer motion. 
This way, that component will be re usable wherever we need it. Then import the same in app/page.tsx

**Comment by Sourabh782** [comment]
Sure sir, just before making pr can u check it?
image

**Review by Sanchitbajaj02 — APPROVED** [review]

**Pull request merged**